grunt/config/replace.js

```diff
@@ -0,0 +1,56 @@
+module.exports = function (grunt, options) {
+
+  var courseDir = options.sourcedir + 'course/';
+
+  if (options.outputdir.indexOf('build/') > -1) {
+    courseDir = options.outputdir + 'course/';
+  }
+
+  var configJson = grunt.file.readJSON(courseDir + 'config.json');
+  var defaultLanguage = configJson._defaultLanguage || 'en';
+  var courseJson = grunt.file.readJSON(courseDir + defaultLanguage + '/course.json');
+
+  // Backwards compatibility for courses missing 'description'
+  if (!courseJson.hasOwnProperty('description')) {
+    courseJson.description = '';
+  }
+
+  // A shim for edge cases where xAPI has not been configured.
+  if (!configJson.hasOwnProperty('_xapi')) {
```

Comment:
I concur that the shim isn't ideal but it ensures that the XML files don't get created with `@@propertyName` in them.